keras_hub/src/models/yolo_v8/non_max_suppression.py

```diff
@@ -0,0 +1,504 @@
+import math
+
+import keras
+from keras import ops
+
+from keras_hub.src import bounding_box
+from keras_hub.src.api_export import keras_hub_export
+
+
+@keras_hub_export("keras_hub.models.yolov8.NonMaxSuppression")
```

Comment:
- Let's avoid a nested `yolov8`module inside `models`, not other model does this.
- Let's be minimal about what we need to make public.
- If this is not general, just call it `keras_hub.models.YOLOV8NonMaxSuppression` I guess.
- If this is general, we could consider adding this as `keras_hub.layers.NonMaxSuppression`.

Reply:
Yes I agree, I have now:
* Removed the nested yolov8 module.
* Removed any exposure of the NonMaxSuppression layer. I believe that there are currently two implementations. This one and the one found in the [Retinanet module](https://github.com/keras-team/keras-hub/blob/c72a4aadd246e593ff5ec9ac5babc70d0e3eb206/keras_hub/src/models/retinanet/non_max_supression.py#L14url). We can in another PR decide which one to reuse and make public.